File: apps/user/src/service/AdminService.ts
'use strict'

import {EmailDao} from "../dao/EmailDao";
import {DeleteAccountRequest} from "../model/request/DeleteAccountRequest";

class AdminService {
    private emailDao: EmailDao
    private readonly adminEmail: string

    constructor() {
        this.emailDao = new EmailDao()
        // @ts-ignore
        this.adminEmail = process.env.ADMIN_EMAIL
    }

    async deleteAccount(request: DeleteAccountRequest) {
        const subject = "Delete user account"
        const body = `Request to delete user ${request.id}`
        await this.emailDao.sendEmail(subject, body, this.adminEmail, null)
    }
}

export {AdminService}

File: apps/user/src/service/admin_service.py
import os

from dao.email_dao import EmailDao
from model.requests import DeleteAccountRequest


class AdminService:
    def __init__(self):
        self.email_dao = EmailDao()
        self.admin_email = os.environ["ADMIN_EMAIL"]

    def delete_account(self, request: DeleteAccountRequest):
        subject = "Delete user account"
        body = f"Request to delete user ${request.id}"

        self.email_dao.send_email(subject=subject, body=body, destination=self.admin_email)
        